src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/X509Certificates/CertificateRevocationListBuilder.Load.cs

```diff
@@ -0,0 +1,286 @@
+
+                AsnValueReader certificateList = reader.ReadSequence();
+                AsnValueReader tbsCertList = certificateList.ReadSequence();
+                AlgorithmIdentifierAsn.Decode(ref certificateList, ReadOnlyMemory<byte>.Empty, out _);
+
+                if (!certificateList.TryReadPrimitiveBitString(out _, out _))
+                {
+                    throw new CryptographicException(SR.Cryptography_Der_Invalid_Encoding);
+                }
+
+                certificateList.ThrowIfNotEmpty();
+
+                int version = 0;
+
+                if (tbsCertList.PeekTag().HasSameClassAndValue(Asn1Tag.Integer))
+                {
+                    // https://datatracker.ietf.org/doc/html/rfc5280#section-5.1 says the only
+                    // version values are v1 (0) and v2 (1).
+                    //
+                    // Since v1 (0) is supposed to not write down the version value, v2 (1) is the
+                    // only legal value to read.
+                    if (!tbsCertList.TryReadInt32(out version) || version != 1)
+                    {
+                        throw new CryptographicException(SR.Cryptography_Der_Invalid_Encoding);
+                    }
+                }
+
+                AlgorithmIdentifierAsn.Decode(ref tbsCertList, ReadOnlyMemory<byte>.Empty, out _);
+                // X500DN
+                tbsCertList.ReadSequence();
+
+                // thisUpdate
+                ReadX509Time(ref tbsCertList);
+
+                // nextUpdate
+                ReadX509TimeOpt(ref tbsCertList);
+
+                AsnValueReader revokedCertificates = tbsCertList.ReadSequence();
+
+                if (version > 0 && tbsCertList.HasData)
+                {
+                    AsnValueReader crlExtensionsExplicit = tbsCertList.ReadSequence(new Asn1Tag(TagClass.ContextSpecific, 0));
+                    AsnValueReader crlExtensions = crlExtensionsExplicit.ReadSequence();
+                    crlExtensionsExplicit.ThrowIfNotEmpty();
+
+                    while (crlExtensions.HasData)
+                    {
+                        AsnValueReader extension = crlExtensions.ReadSequence();
+                        Oid? extnOid = Oids.GetSharedOrNullOid(ref extension);
+
+                        if (extnOid is null)
+                        {
+                            extension.ReadObjectIdentifier();
+                        }
+
+                        if (extension.PeekTag().HasSameClassAndValue(Asn1Tag.Boolean))
+                        {
+                            extension.ReadBoolean();
+                        }
+
+                        if (!extension.TryReadPrimitiveOctetString(out ReadOnlySpan<byte> extnValue))
+                        {
+                            throw new CryptographicException(SR.Cryptography_Der_Invalid_Encoding);
+                        }
+
+                        if (ReferenceEquals(extnOid, Oids.CommonNameOid))
```

Comment:
I'm not sure why `ReferenceEquals` or `Oids.CommonNameOid` is being used here. Should this be `Oids.ValueEquals` and `Oids.CrlNumberOid`?

Reply:
CommonNameOid was a bad expansion of CNO, residual from the lack of having written tests yet :).

ReferenceEquals is valid because of GetSharedOrNullOid (and not caring about "any other value" the way we do in the X500 DN enumerator).  Added a comment.